src/backend/distributed/planner/distributed_planner.c

```diff
@@ -117,17 +123,28 @@ static bool HasUnresolvedExternParamsWalker(Node *expression, ParamListInfo boun
 static bool IsLocalReferenceTableJoin(Query *parse, List *rangeTableList);
 static bool QueryIsNotSimpleSelect(Node *node);
 static bool UpdateReferenceTablesWithShard(Node *node, void *context);
+static PlannedStmt * FastPastDistributedStmt(DistributedStmtOptions *opts,
+											 Const *distributionKeyValue);
+static PlannedStmt * NonFastPathDistributedStmt(DistributedStmtOptions *opts,
+												List *rangeTableList, int rteIdCounter);
+static PlannedStmt * NonDistributedStmt(DistributedStmtOptions *opts);
 
 /* Distributed planner hook */
 PlannedStmt *
 distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 {
 	PlannedStmt *result = NULL;
 	bool needsDistributedPlanning = false;
-	Query *originalQuery = NULL;
 	bool setPartitionedTablesInherited = false;
 	List *rangeTableList = ExtractRangeTableEntryList(parse);
 	int rteIdCounter = 1;
+	bool fastPathRouterQuery = false;
+	Const *distributionKeyValue = NULL;
+	DistributedStmtOptions opts = {
+		.parse = parse,
+		.cursorOptions = cursorOptions,
+		.boundParams = boundParams,
+	};
 
```

Comment:
To clean things up a bit further, maybe we can have an early return for `if (!CitusHasBeenLoaded())`?

Reply:
I'm not sure what you mean with this, what would we return in that case?